src/render/decorations/CHyprGroupBarDecoration.cpp

```diff
@@ -314,13 +314,14 @@ bool CHyprGroupBarDecoration::onEndWindowDragOnDeco(CWindow* pDraggedWindow, con
     const int   WINDOWINDEX  = BARRELATIVEX < 0 ? -1 : (BARRELATIVEX) / (m_fBarWidth + BAR_HORIZONTAL_PADDING);
 
     CWindow*    pWindowInsertAfter = m_pWindow->getGroupWindowByIndex(WINDOWINDEX);
+    CWindow*    pDraggedHead       = pDraggedWindow->m_sGroupData.pNextWindow ? pDraggedWindow->getGroupHead() : pDraggedWindow;
 
-    g_pLayoutManager->getCurrentLayout()->onWindowRemoved(pDraggedWindow);
+    g_pLayoutManager->getCurrentLayout()->onWindowRemovedTiling(pDraggedWindow);
```

Comment:
wait a second, dragged windows are floating?

Reply:
that's used after window gets placed by the layout to remove the node
`onWindowRemoved` was removing grouped properties and messing stuff up
`onWindowRemovedTiling` does everything the same except removing the grouped properties
so now everything seems to work properly

Reply:
I don't like this. This may silently break very easily in the future.

Reply:
I don't like it either, but we can't access the layout from the outside to delete the nodes

Reply:
we can save the group state tho

Reply:
we could but we would still be using the same function (wrapped inside another one)
not sure if you mean the unnecessary placing and removing of windows or bypassing `onWindowRemoved()`

Reply:
I'd want to call `onWindowRemoved` instead